src/KubernetesClient/Informers/Controller.cs

```diff
@@ -0,0 +1,122 @@
+using k8s.cache;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace k8s.informers
+{
+    public class Controller<T, L> : IController<T, L>
+    {
+        public IReflector<T,L> _reflector;       
+        public IStore _store;
+        private Queue<Tuple<WatchEventType, IKubernetesObject>> _fifo;
+        private AutoResetEvent _fifoEvent;
+        private Action<WatchEventType, IKubernetesObject> _onAdd;
+        private Action<WatchEventType, IKubernetesObject> _onDelete;
+        private Action<WatchEventType, IKubernetesObject, IKubernetesObject> _onUpdate;
+
+        private void onAdd(WatchEventType eventType, IKubernetesObject obj) {
+            if (_onAdd != null) {
+                _onAdd(eventType, obj);
+            }
+        }
+
+        private void onDelete(WatchEventType eventType, IKubernetesObject deletedObj) {
+            if (_onDelete != null) {
+                _onDelete(eventType, deletedObj);
+            }
+        }
+
+        private void onUpdate(WatchEventType eventType, IKubernetesObject oldObj, IKubernetesObject newObj) {
+            if (_onUpdate != null) {
+                _onUpdate(eventType, oldObj, newObj);
+            }
+        }
+
+        public Controller(ListerWatcher<T,L> lw, IStore s) 
+        {            
+            _store = s;
+            _fifo = new Queue<Tuple<WatchEventType, IKubernetesObject>>();
+            _fifoEvent = new AutoResetEvent(false);
+            _reflector = new Reflector<T, L>(lw, TimeSpan.FromMinutes(1), _fifo, _fifoEvent);
+            
+        }
+
+        /// <summary>
+        /// Installs resource handler call backs for add, del and update of objects.
+        /// </summary>
+        public void AddResourceHandlers(Action<WatchEventType, IKubernetesObject> onAdd, Action<WatchEventType, IKubernetesObject> onDelete, 
```

Comment:
Per above "AddResourceHandler" implies that it would be added, when in fact it replaces an existing handler.